api/grpc/server.go

```diff
@@ -697,6 +714,13 @@ func toGrpcError(err perun.APIErrorV2) *pb.MsgErrorV2 {
 				Requirement: info.Requirement,
 			},
 		}
+	case perun.ErrV2InfoFailedPreConditionSessionClose:
+
+		grpcErr.AddInfo = &pb.MsgErrorV2_ErrV2InfoFailedPreConditionSessionClose{
+			ErrV2InfoFailedPreConditionSessionClose: &pb.ErrV2InfoFailedPreConditionSessionClose{
+				ChInfo: toGrpcChsInfo(info.OpenChsInfo),
```

Comment:
We should be returning a "payment channels info" instead of "channels info". To do this, you will have to convert the "channels info" in the `error.AddInfo` field to "payment channels info" in the `app/payment` package.

After that, in `grpc` wrapper, you will be having "payment channels info" which you can process using the already existing `toGrpcPayChsInfo` function.